enterprise_access/apps/bffs/tests/test_builders.py

```diff
@@ -0,0 +1,161 @@
+import ddt
+from django.test import RequestFactory, TestCase
+
+from enterprise_access.apps.api_client.tests.test_license_manager_client import TestLicenseManagerUserApiClient
+from enterprise_access.apps.bffs.context import HandlerContext
+from enterprise_access.apps.bffs.response_builder import BaseLearnerResponseBuilder, BaseResponseBuilder
+from enterprise_access.apps.bffs.tests.test_handlers import TestBaseHandlerMixin
+
+
+@ddt.ddt
+class TestBaseResponseBuilder(TestBaseHandlerMixin):
+    def setUp(self):
+        super().setUp()
+        self.base_response_builder = BaseResponseBuilder(self.context)
+
+    def test_base_build_error(self):
+        with self.assertRaises(NotImplementedError):
+            self.base_response_builder.build()
+
+    @ddt.data(
+        {
+            'errors': True,
+            'warnings': True,
+        },
+        {
+            'errors': True,
+            'warnings': False,
+        },
+        {
+            'errors': False,
+            'warnings': True,
+        },
+        {
+            'errors': False,
+            'warnings': False,
+        }
+    )
+    @ddt.unpack
+    def test_add_errors_warnings_to_response(self, errors, warnings):
+        sample_response = {
+            "some_existing_metadata": "should_still_exist"
+        }
+        expected_output = {
+            **sample_response,
+        }
+        if errors:
+            self.context.add_error(
+                **self.mock_error
+            )
+            expected_output['errors'] = [self.mock_error]
+        if warnings:
+            self.context.add_warning(
+                **self.mock_warning
+            )
+            expected_output['warnings'] = [self.mock_warning]
+        response_data = self.base_response_builder.add_errors_warnings_to_response(sample_response)
+        self.assertEqual(response_data, expected_output)
+
+    @ddt.data(
+        {
+            'status_code': 201
+        },
+        {
+            'status_code': None
+        }
+    )
+    @ddt.unpack
+    def test_get_status_code(self, status_code):
```

Comment:
nit: might be worth dropping a `TODO` comment referencing https://2u-internal.atlassian.net/browse/ENT-9633, as  if we do add the `status_code` class attribute in `HandlerContext`, it may come with a minor change or two to this test case.